python/cucim/src/cucim/skimage/filters/thresholding.py

```diff
@@ -366,6 +366,9 @@ def threshold_otsu(image=None, nbins=256, *, hist=None):
             return first_pixel
 
     counts, bin_centers = _validate_image_histogram(image, hist, nbins)
+    # small size counts, bin_centers -> faster on the host
+    counts = cp.asnumpy(counts)
+    bin_centers = cp.asnumpy(bin_centers)
```

Comment:
@jakirkham, as a concrete example of what you were asking, now that we have computed the histogram and transfered back to the host, we could replace the rest of this function with a call to the skimage function:

```Python
return skimage.threshold_otsu(image=None, nbins=nbins, hist=(counts, bin_centers))
```
It does look like the `hist` argument has been available since scikit-image 0.18

Reply:
I have now removed redundant code and call the scikit-image functions with the histogram that was computed on the GPU. See commit fc73695